src/Node/HTTP/Secure.purs

```diff
@@ -90,16 +90,17 @@ import Unsafe.Coerce (unsafeCoerce)
 
 -- | Create an HTTPS server, given the SSL options and a function to be executed
 -- | when a request is received.
-foreign import createServerImpl ::
-  Foreign ->
-  (Request -> Response -> Effect Unit) ->
-  Effect Server
+foreign import createServerImpl
+  :: Foreign
+  -> (Request -> Response -> Effect Unit)
+  -> Effect Server
 
 -- | Create an HTTPS server, given the SSL options and a function to be executed
 -- | when a request is received.
-createServer :: Options SSLOptions ->
-                (Request -> Response -> Effect Unit) ->
-                Effect Server
+createServer
+  :: Options SSLOptions
+  -> (Request -> Response -> Effect Unit)
+  -> Effect Server
```

Comment:
> New function `HTTP.onRequest`. Solves HTTP.createServer then close after one connection (#46).

This comment also applies to the `HTTP.purs` file. 
I think we should add a variant of `createServer` (e.g. `createServer'`) that drops the `(req, res) => ...` `requestListener` function. From the [Node docs (very bottom of section)](https://nodejs.org/docs/latest/api/http.html#httpcreateserveroptions-requestlistener), the `requestListener` function optional argument, when present, is added automatically to the `onRequest` callback. So...

```javascript
// This...
const server = http.createServer((request, response) => { ... });
// is the same as
const server = http.createServer();
server.on("request", (request, response) => { ... });
```
Thus:
```purs
-- Rather than this PR's current code in the test:
server <- createServer \_ _ -> pure unit`
onRequest server ...
-- it could be
server <- createServer'
onRequest server ...
```

Reply:
Also, perhaps here we should mention in the docs what the Node docs say?
> Since there are no browsers known that support [unencrypted HTTP/2](https://http2.github.io/faq/#does-http2-require-encryption), the use of [http2.createSecureServer()](https://nodejs.org/docs/latest/api/http2.html#http2createsecureserveroptions-onrequesthandler) is necessary when communicating with browser clients.

Reply:
That would be good. Can we do it in a separate PR? I only made the minimum changes to the __HTTP__ module to make the tests work and consistently terminate.